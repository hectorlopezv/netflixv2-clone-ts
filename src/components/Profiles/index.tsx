import React from 'react';
import {Container, Title, Picture, Name, Item, List} from './styles/Profilex';



export interface ProfilesProps {
    
}
 
const Profiles: React.FC<ProfilesProps> = ({children, ...props}) => {
    return ( 
        <Container {...props}>{children}</Container>
     );
}
 
export default Profiles;

export interface Profiles_TitleProps {
    
}


export const Profiles_Title: React.FC<Profiles_TitleProps> = ({children, ...props}) => {
    return ( 
        <Title {...props}>{children}</Title>
     );
}
 

export interface Profiles_PictureProps {
    src: string;
}
 

export const Profiles_Picture: React.FC<Profiles_PictureProps> = ({src, ...props}) => {
    //reconsiliation
    return ( 
        <Picture {...props} src={src ? `/images/users/${src}.png` :`/images/misc/loading.gif`} />
     );
}
 

export interface Profiles_NameProps {
    
}
 

export const Profiles_Name: React.FC<Profiles_NameProps> = ({children, ...props}) => {
    return ( 
        <Name {...props}>{children}</Name>
     );
}
 

export interface Profiles_UserProps {
    onClick: any;
}
 

export const Profiles_User: React.FC<Profiles_UserProps> = ({children, ...props}) => {
    return ( 
        <Item {...props}>{children}</Item>
     );
}


export interface Profiles_ListProps {
    
}
 
export const Profiles_List: React.FC<Profiles_ListProps> = ({children, ...props}) => {
    return ( 
        <List {...props}>{children}</List>
     );
}
 
