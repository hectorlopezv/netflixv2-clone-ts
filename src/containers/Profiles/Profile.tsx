/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import {} from 'react-router-dom';
import * as ROUTES from '../../Routes';
import logo from '../Header/logo.svg';
import Header, {Header_Frame, Header_Logo } from '../../components/Header';
import Profiles, {Profiles_Title, Profiles_List,Profiles_User, Profiles_Picture, Profiles_Name } from '../../components/Profiles';

export interface SelectProfileContainerProps {
    user?: any;
    setProfile?: any;
}
 
const SelectProfileContainer: React.FC<SelectProfileContainerProps> = ({user, setProfile}) => {
    //ACCOUNT CAN HAVE MULTIPLE USER!
    //1 STEP
    return (
        <>
        <Header bg = {false}>
            <Header_Frame>
                <Header_Logo to={ROUTES.HOME} alt="Netflix" src={logo} />
            </Header_Frame>
        </Header>

        <Profiles>
            <Profiles_Title>Who is Watching</Profiles_Title>
            <Profiles_List>
                <Profiles_User onClick={() => setProfile({
                        displayName: user.displayName,
                        photoURL: user.photoURL
                    })}>
                    <Profiles_Picture src={user?.photoURL} />
                    <Profiles_Name>{user?.displayName}</Profiles_Name>
                </Profiles_User>
            </Profiles_List>
        </Profiles>
        </>
       
    );
}
 
export default SelectProfileContainer;