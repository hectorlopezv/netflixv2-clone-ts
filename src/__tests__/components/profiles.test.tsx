/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { render, fireEvent, getByTestId, getByAltText } from '@testing-library/react';

import Profiles, {Profiles_Title, Profiles_List,Profiles_User, Profiles_Picture, Profiles_Name } from '../../components/Profiles';

describe('Profile component', () => {
    it('it renders <Profiles /> Component with populated', () => {
        const {container, getByText, getByTestId} = render(       
            <Profiles>
                <Profiles_Title>Who is Watching</Profiles_Title>
                <Profiles_List>
                    <Profiles_User onClick={() => {}}>
                        <Profiles_Picture src={"/images/karl.png"} data-testid="profile-picture" />
                        <Profiles_Name>Karl Hadwen</Profiles_Name>
                    </Profiles_User>
                </Profiles_List>
            </Profiles>
        );
        //assertions
        expect(getByText('Who is Watching')).toBeTruthy();
        expect(getByTestId('profile-picture')).toBeTruthy();
        expect(getByText("Karl Hadwen")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});