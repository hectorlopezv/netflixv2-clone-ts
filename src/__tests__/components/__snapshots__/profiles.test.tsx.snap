// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component it renders <Profiles /> Component with populated 1`] = `
<div
  class="Profilex__Container-sc-7rq9xm-0 kQeQMr"
>
  <h1
    class="Profilex__Title-sc-7rq9xm-1 gfhjgh"
  >
    Who is Watching
  </h1>
  <ul
    class="Profilex__List-sc-7rq9xm-4 iVSQRF"
  >
    <li
      class="Profilex__Item-sc-7rq9xm-5 ciqsgI"
    >
      <img
        class="Profilex__Picture-sc-7rq9xm-2 jWDsAp"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="Profilex__Name-sc-7rq9xm-3 duiyaP"
      >
        Karl Hadwen
      </p>
    </li>
  </ul>
</div>
`;
